### ISSUE TITLE
bug: sendMsg function doesn't always validate length

### ISSUE THREAD

**Issue by theckman** [DataDog/datadog-go#24]
Hey there,

I was doing some reading of the codebase and I think stumbled upon a bug in the `sendMsg` method. Specifically, it seems that the length of the message is only checked if the client is in a buffered mode:

* https://github.com/DataDog/datadog-go/blob/426ff2e0daab1153de70993f85773d91302d50a0/statsd/statsd.go#L224-L228

Looking at it further, it seems the comment above the length check is at odds with what the code actually codes:

* https://github.com/DataDog/datadog-go/blob/426ff2e0daab1153de70993f85773d91302d50a0/statsd/statsd.go#L225

It's hard to know what the expected behavior truly is because the tests aren't currently ran for the function because it's unexported. Even then it seems to be at adds with the code itself:

* https://github.com/DataDog/datadog-go/blob/426ff2e0daab1153de70993f85773d91302d50a0/statsd/statsd_test.go#L341-L351

If you export the function and try to run the test, it results in a panic which is probably why it is unexported:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x7ea3b]

goroutine 26 [running]:
panic(0x17df00, 0xc4200140c0)
        /usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc42009c480)
        /usr/local/Cellar/go/1.7.4/libexec/src/testing/testing.go:579 +0x25d
panic(0x17df00, 0xc4200140c0)
        /usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:458 +0x243
github.com/DataDog/datadog-go/statsd.(*Client).flush(0xc4200ae2a0, 0x0, 0x0)
        /Users/theckman/go/src/github.com/DataDog/datadog-go/statsd/statsd.go:201 +0xdb
github.com/DataDog/datadog-go/statsd.(*Client).append(0xc4200ae2a0, 0xc4200a8c00, 0x598, 0xc4200a8c00, 0x598)
        /Users/theckman/go/src/github.com/DataDog/datadog-go/statsd/statsd.go:141 +0xab
github.com/DataDog/datadog-go/statsd.(*Client).sendMsg(0xc4200ae2a0, 0xc4200a8c00, 0x598, 0x0, 0x0)
        /Users/theckman/go/src/github.com/DataDog/datadog-go/statsd/statsd.go:229 +0x124
github.com/DataDog/datadog-go/statsd.TestSendMsg(0xc42009c480)
        /Users/theckman/go/src/github.com/DataDog/datadog-go/statsd/statsd_test.go:343 +0xee
testing.tRunner(0xc42009c480, 0x1be4e8)
        /usr/local/Cellar/go/1.7.4/libexec/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
        /usr/local/Cellar/go/1.7.4/libexec/src/testing/testing.go:646 +0x2ec
exit status 2
```

In short, I think the `sendMsg` method should check the length of the message whether or not it's a buffered client. Secondly, I think either the function's logic needs to be updated or the comment/tests need to be updated to match the expected behavior. Lastly, I think the changes in #22 should also be applied to `sendMsg()` / `append()`.
